include/oneapi/dpl/pstl/execution_defs.h

```diff
@@ -228,6 +228,9 @@ struct __range_traits
 template <typename _R>
 using __value_t = typename __range_traits<_R>::__value_t;
 
+template <typename _Proj, typename _Range>
+using __key_t = decltype(::std::declval<_Proj>()(::std::declval<_Range&>()[0]));
```

Comment:
Seems like for `identity` it would return (for example) `int&&` and later our `is_arithmetic` check would be `false` that eliminates possibility to use radix sort for suitable types. Therefore:

```suggestion
using __key_t = std::decay_t<decltype(::std::declval<_Proj&>()(::std::declval<_Range&>()[0]))>;
```